### PR TITLE
Fix use of deprecated api and compiler warnings

### PR DESCRIPTION
**Pull request by afh**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -20,7 +20,10 @@ endif()
 
 enable_testing()
 
-add_definitions(-std=c++11)
+add_definitions(
+  -std=c++11
+  -DBOOST_FILESYSTEM_NO_DEPRECATED
```
Comment (afh) [inline CMakeLists.txt:25]: For details regarding the added `-DBOOST_FILESYSTEM_NO_DEPRECATED` definition see [Boost Filesystem Deprecated Features](https://www.boost.org/doc/libs/1_81_0/libs/filesystem/doc/deprecated.html)

File: src/timelog.cc [hunk 2]
```diff
@@ -55,7 +55,7 @@ namespace {
       curr->append_note(in_event.note.c_str(), *context.scope);
 
     char buf[32];
-    std::sprintf(buf, "%lds", long((out_event.checkin - in_event.checkin)
+    std::snprintf(buf, 32, "%lds", long((out_event.checkin - in_event.checkin)
```
Comment (afh) [inline src/timelog.cc:58]: According the [adfad](https://en.cppreference.com/w/cpp/io/c/fprintf) `snprintf` is available in C++11, which Ledger already requires (see [`CMakeLists.txt:23`](https://github.com/ledger/ledger/blob/master/CMakeLists.txt#L23))

File: src/utils.cc [hunk 3]
```diff
@@ -849,7 +849,11 @@ path resolve_path(const path& pathname)
   path temp = pathname;
   if (temp.string()[0] == '~')
     temp = expand_path(temp);
+#if (BOOST_VERSION >= 106000)
+  temp.lexically_normal();
```
Comment (afh) [inline src/utils.cc:853]: `lexically_normal` was introduced in Boost 1.60.0 (see [**New Features** section of Boost 1.60.0 Release Notes](https://www.boost.org/users/history/version_1_60_0.html)) so as long as Ledger maintains support for Boost <=1.60.0 (currently [1.49](https://github.com/ledger/ledger/blob/master/CMakeLists.txt#L89)) use of `lexically_normal` needs to be version guarded.

**Review by tbm — APPROVED** [review]
thanks!

**Comment by afh** [comment]
Thanks for reviewing and mergin, @tbm 🙏

**Pull request merged**